src/annotator.cpp

```diff
@@ -1156,200 +1155,259 @@ std::string Annotator::AnnotatorImpl::makeUniqueId()
 std::string Annotator::AnnotatorImpl::id(const AnyCellmlElementPtr &item)
 {
     std::string id;
-    CellmlElementType type = item->type();
-    if (type == CellmlElementType::COMPONENT) {
+    switch (item->type()) {
+    case CellmlElementType::COMPONENT:
         id = item->component()->id();
-    } else if (type == CellmlElementType::COMPONENT_REF) {
+        break;
+    case CellmlElementType::COMPONENT_REF:
         id = item->component()->encapsulationId();
-    } else if (type == CellmlElementType::CONNECTION) {
+        break;
+    case CellmlElementType::CONNECTION: {
         auto variablePair = item->variablePair();
         id = Variable::equivalenceConnectionId(variablePair->variable1(), variablePair->variable2());
-    } else if (type == CellmlElementType::ENCAPSULATION) {
+    } break;
+    case CellmlElementType::ENCAPSULATION:
         id = item->model()->encapsulationId();
-    } else if (type == CellmlElementType::IMPORT) {
+        break;
+    case CellmlElementType::IMPORT:
         id = item->importSource()->id();
-    } else if (type == CellmlElementType::MAP_VARIABLES) {
+        break;
+    case CellmlElementType::MAP_VARIABLES: {
         auto variablePair = item->variablePair();
         id = Variable::equivalenceMappingId(variablePair->variable1(), variablePair->variable2());
-    } else if (type == CellmlElementType::MODEL) {
+    } break;
+    case CellmlElementType::MODEL:
         id = item->model()->id();
-    } else if (type == CellmlElementType::RESET) {
+        break;
+    case CellmlElementType::RESET:
         id = item->reset()->id();
-    } else if (type == CellmlElementType::RESET_VALUE) {
+        break;
+    case CellmlElementType::RESET_VALUE:
         id = item->reset()->resetValueId();
-    } else if (type == CellmlElementType::TEST_VALUE) {
+        break;
+    case CellmlElementType::TEST_VALUE:
         id = item->reset()->testValueId();
-    } else if (type == CellmlElementType::UNIT) {
+        break;
+    case CellmlElementType::UNIT: {
         auto unitItem = item->unitsItem();
         id = unitItem->units()->unitId(unitItem->index());
```

Comment:
Reduces readability.

Reply:
Actually, my comment was wrong. If anything, it should have been `id = item->unitsItem()->units()->unitId(unitItem->index());` since your original `unitItem` is used twice, but this would indeed make it unnecessarily difficult to read.